app/my_dict.py

```diff
@@ -0,0 +1,122 @@
+        self.hash_table: list[Node | None] = [None] * self.capacity
+
+    def _calculate_index(self, key: Hashable) -> int:
+        index = hash(key) % self.capacity
+
+        while (self.hash_table[index] is not None
+               and self.hash_table[index].key != key):
+            index = (index + 1) % self.capacity
+
+        return index
+
+    @property
+    def current_max_size(self) -> float:
+        return self.capacity * RESIZE_THRESHOLD
+
+    def resize(self) -> None:
+        old_hash_table = self.hash_table
+        new_size = self.capacity * 2
+
+        self.__init__(new_size)
+
+        for node in old_hash_table:
+            if node is not None:
+                self.__setitem__(node.key, node.value)
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self._calculate_index(key)
+
+        if self.hash_table[index] is None:
+            if self.size + 1 >= self.current_max_size:
+                self.resize()
+                return self.__setitem__(key, value)
+
+            self.size += 1
+
+        self.hash_table[index] = Node(key, value)
+
+    def __getitem__(self, key: Hashable) -> Any:
+        index = self._calculate_index(key)
+
+        if self.hash_table[index] is None:
+            raise KeyError(f"Cannot find value for key: {key}")
+
+        return self.hash_table[index].value
+
+    def __len__(self) -> int:
+        return self.size
+
+    def clear(self) -> None:
+        self.hash_table = [None] * self.capacity
+        self.size = 0
+
+    def __delitem__(self, key: Hashable) -> None:
+        index = self._calculate_index(key)
+
+        if self.hash_table[index] is None:
+            raise KeyError(f"Cannot find value for key: {key}")
+
+        self.hash_table[index] = None
+        self.size -= 1
+
+    def get(self, key: Hashable, default: Any = None) -> Any:
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
+    def pop(self, key: Hashable, default: Any = None) -> None:
+        try:
+            del self[key]
+        except KeyError:
+            return default
+
+    def update(self, key: Hashable, value: Any) -> None:
+        index = self._calculate_index(key)
+        step = 1
+
+        while self.hash_table[index] is not None:
+            if self.hash_table[index].key == key:
+                self.hash_table[index].value = value
+                return
+            index = self._calculate_index(key)
+            step += 1
+
+        self.hash_table[index] = Node(key, value)
+        self.size += 1
+
+    def __iter__(self) -> list:
+        yield self.hash_table
+
+    @property
+    def pairs(self) -> list:
+        return [(key, self[key]) for key in self.hash_table]
+
+    def __eq__(self, other: Dictionary) -> bool:
+        if self is other:
+            return True
+        if type(self) is not type(other):
+            return False
+        return set(self.pairs) == set(other.pairs)
```

Comment:
The `__eq__` method should iterate over the hash table to compare each key-value pair properly. Ensure that both dictionaries have the same size and that all key-value pairs match.